src/main/java/com/autotune/analyzer/services/DSMetadataService.java

```diff
@@ -90,8 +91,25 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
 
             DataSourceInfo datasource = new ExperimentDBService().loadDataSourceFromDBByName(dataSourceName);
             if(null != datasource) {
-                new DataSourceManager().importMetadataFromDataSource(datasource);
                 DataSourceMetadataInfo dataSourceMetadata = new ExperimentDBService().loadMetadataFromDBByName(dataSourceName, "false");
+                if (null != dataSourceMetadata) {
+                    ValidationOutputData validationOutputData = new ExperimentDAOImpl().deleteKruizeDSMetadataEntryByName(dataSourceName);
+
+                    if (validationOutputData.isSuccess()) {
+                        new DataSourceManager().deleteMetadataFromDataSource(datasource);
+                    } else {
+                        String errorMessage = validationOutputData.getMessage();
+                        sendErrorResponse(
+                                response,
+                                new Exception(AnalyzerErrorConstants.APIErrors.DSMetadataAPI.MISSING_DATASOURCE_METADATA_EXCPTN),
+                                HttpServletResponse.SC_BAD_REQUEST,
+                                errorMessage
+                        );
+                    }
+                }
+
+                new DataSourceManager().importMetadataFromDataSource(datasource);
```

Comment:
As mentioned in the review, this can cause DB corruption if for any reason the import fails as we would have removed the DB content. Please reverse the order and add validation to the import results

Reply:
Updated the code flow including validation of imported `DataSourceMetadataInfo` object